packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -0,0 +1,667 @@
+
+        save("ProxyAdmin", address(admin));
+        console.log("ProxyAdmin deployed at %s", address(admin));
+        return address(admin);
+    }
+
+    /// @notice Deploy the L1StandardBridgeProxy
+    function deployL1StandardBridgeProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        L1ChugSplashProxy proxy = new L1ChugSplashProxy(proxyAdmin);
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("L1StandardBridgeProxy", address(proxy));
+        console.log("L1StandardBridgeProxy deployed at %s", address(proxy));
+        return address(proxy);
+    }
+
+    /// @notice Deploy the L2OutputOracleProxy
+    function deployL2OutputOracleProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("L2OutputOracleProxy", address(proxy));
+        console.log("L2OutputOracleProxy deployed at %s", address(proxy));
+        return address(proxy);
+    }
+
+    /// @notice Deploy the L1CrossDomainMessengerProxy
+    function deployL1CrossDomainMessengerProxy() broadcast() public returns (address) {
+        AddressManager addressManager = AddressManager(mustGetAddress("AddressManager"));
+        string memory contractName = "OVM_L1CrossDomainMessenger";
+        ResolvedDelegateProxy proxy = new ResolvedDelegateProxy(addressManager, contractName);
+
+        save("L1CrossDomainMessengerProxy", address(proxy));
+        console.log("L1CrossDomainMessengerProxy deployed at %s", address(proxy));
+
+        address addr = addressManager.getAddress(contractName);
+        if (addr != address(proxy)) {
+            addressManager.setAddress(contractName, address(proxy));
+        }
+
+        require(addressManager.getAddress(contractName) == address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the OptimismPortalProxy
+    function deployOptimismPortalProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("OptimismPortalProxy", address(proxy));
+        console.log("OptimismPortalProxy deployed at %s", address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the OptimismMintableERC20FactoryProxy
+    function deployOptimismMintableERC20FactoryProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy(proxyAdmin);
```

Comment:
Should we also use named args here? ie
```suggestion
        Proxy proxy = new Proxy({
            _admin: proxyAdmin
        });
```

Reply:
Yes good call, I def think we should lean towards named arg syntax as much as possible to decrease the cognitive overhead of reading the code and knowing which arg is which.